src/VisualStudio/Core/Def/LanguageService/AbstractLanguageService`2.cs

```diff
@@ -56,10 +58,10 @@ internal abstract partial class AbstractLanguageService<TPackage, TLanguageServi
     // Note: The lifetime for state in this class is carefully managed.  For every bit of state
     // we set up, there is a corresponding tear down phase which deconstructs the state in the
     // reverse order it was created in.
-    internal EditorOptionsService EditorOptionsService { get; private set; }
-    internal VisualStudioWorkspaceImpl Workspace { get; private set; }
-    internal IVsEditorAdaptersFactoryService EditorAdaptersFactoryService { get; private set; }
-    internal AnalyzerFileWatcherService AnalyzerFileWatcherService { get; private set; }
+    internal readonly EditorOptionsService EditorOptionsService;
```

Comment:
>internal readonly EditorOptionsService EditorOptionsService; [](http://example.com/codeflow?start=4&length=60)

by init'ing these in the ctor and not nulling them out, could this file be made nullable without too much effort now?

Reply:
yes.  will do in followup.